ros2_ws/src/otto_deliberation/src/tasks.cpp

```diff
@@ -241,4 +145,69 @@ Task RobotTasks::ConstructTask(const RobotTaskKey& taskKey) const
 bool RobotTasks::GetIfTaskNeedsLock(const RobotTaskKey& taskName) const
 {
     return m_acquireLock.count(taskName) > 0;
-}
\ No newline at end of file
+}
+
+bool RobotTasks::ValidateTaskSet(const RobotTaskSet& taskSet, const std::string& taskSetName) const
+{
+    bool isValid = true;
+    for (const auto& task : taskSet)
+    {
+        if (m_validTasks.count(task) == 0)
+        {
+            isValid = false;
+            RCLCPP_ERROR(rclcpp::get_logger("rclcpp"), "Task %s is in %s but not in tasks list", task.c_str(), taskSetName.c_str());
+        }
+    }
+    return isValid;
```

Comment:
I believe that there is no break after first error exactly because of logs - listing all the tasks that are not valid. I would stick to this approach, especially when ValidateTaskSet is called only once, on deliberation initialization.

.contains() was introduced in C++20. Unfortunately we do not have it here. Luckily, .contains() and .count() have the same complexity.